### ISSUE TITLE
Windows bin proxy for bat file produces incorrect path

### ISSUE THREAD

**Issue by Rarst** [composer/composer#2040]
I am trying to add Windows bat bootstrap script to project which has original script in sh (and so unsuitable for native Windows console).

This is the `wp.bat` script added to project's `bin` folder and `bin` section in `composer.json`:

``` bat
@ECHO OFF
php "%~dp0../php/boot-fs.php" %*
```

Pretty basic - up a level, then down to PHP bootstrap file. It works.

Now when I use project as dependency, Composer produces following proxy script in `vendor/bin`:

``` bat
@ECHO OFF
SET BIN_TARGET=%~dp0\"../wp-cli/wp-cli/bin"\wp.bat
call "%BIN_TARGET%" %*
```

It doesn't work with:

```
Could not open input file: C:\server\ext\wp-cli-test\vendor\bin\.\wp-cli\wp-cli\bin\../php/boot-fs.php
```

The raw path before getting passed to `call` seems to be:

```
C:\server\ext\wp-cli-test\vendor\bin\\"../wp-cli/wp-cli/bin"\wp.bat
```

It seems to lose one dot in the middle there. After first part of the path, ending in `\vendor\bin\`.
- if I remove quotes from `call` line around `%BIN_TARGET%` it seems to work
- same if I remove quotes from _inside_ `BIN_TARGET` value

Quotes inside seem to come from `escapeshellarg()`, maybe that is too much?..

TLDR (now that it is not 4am):

[`LibraryInstaller:280` ](https://github.com/composer/composer/blob/master/src/Composer/Installer/LibraryInstaller.php#L280) uses [`escapeshellarg()`](http://php.net/manual/en/function.escapeshellarg.php) on part of a generated and concatenated file path. On Windows platform this wraps that part in double quotes, which (inside of the path) is confusing and seems to lead to quote mismatches and path issues.

Issue context https://github.com/wp-cli/wp-cli/issues/542
Pull request context https://github.com/wp-cli/wp-cli/pull/544/files
